### PR TITLE
Register metric to get datetime of last solver synced in database

### PR DESCRIPTION
**Pull request by mayaCostantini**
## Related Issues and Dependencies

Related to #822 

## This introduces a breaking change

- No

## This Pull Request implements

Register a metric to get the datetime of the last solver run synced in the database

### CODE REVIEW

**Comment by KPostOffice** [comment]
Depends on: https://github.com/thoth-station/storages/pull/2569

**Comment by mayaCostantini** [comment]
/unhold
https://github.com/thoth-station/storages/pull/2569 was merged

**Comment by mayaCostantini** [comment]
@pacospace PTAL 👍🏻

File: thoth/metrics_exporter/jobs/db.py [hunk 1]
```diff
@@ -123,3 +125,14 @@ def check_is_schema_up2date_for_components(cls) -> None:
...
+    @classmethod
+    @register_metric_job
+    def set_last_solver_datetime(
+        cls, os_name: Optional[str] = None, os_version: Optional[str] = None, python_version: Optional[str] = None
+    ) -> None:
+        """Get datetime of the last solver synced in the database."""
+        last_solver_datetime = cls.graph().get_last_solver_datetime(
+            os_name=os_name, os_version=os_version, python_version=python_version
+        )
+        metrics.graphdb_last_solver_datetime.set(format_datetime(last_solver_datetime))
```
Comment (pacospace) [inline thoth/metrics_exporter/jobs/db.py:138]: `metrics.graphdb_last_solver_datetime.labels(format_datetime(last_solver_datetime)).inc()`

File: thoth/metrics_exporter/metrics.py [hunk 2]
```diff
@@ -98,6 +98,9 @@
 
 graphdb_alembic_table_check = Gauge("thoth_graphdb_alembic_table_check", "alembic version table rows check.", [])
 
+graphdb_last_solver_datetime = Gauge(
+    "thoth_graphdb_last_solver_datetime", "Datetime of the last solver synced in the database.", []
```
Comment (pacospace) [inline thoth/metrics_exporter/metrics.py:102]: add `datetime` label

File: thoth/metrics_exporter/jobs/db.py [hunk 3]
```diff
@@ -18,12 +18,14 @@
...
 
 import logging
+from typing import Optional
 
 import thoth.metrics_exporter.metrics as metrics
 
 from .base import register_metric_job
 from .base import MetricsBase
 from ..configuration import Configuration
+from thoth.common import format_datetime
```
Comment (pacospace) [inline thoth/metrics_exporter/jobs/db.py:28]: please check format is the same between the query and the method in thoth-common

Comment (mayaCostantini) [reply thoth/metrics_exporter/jobs/db.py:28]: `get_last_solver_datetime` returns a `datetime` object and `format_datetime` takes a `datetime` and returns a `str`, so the format should be good for setting the label

[the remark this answers is not in the document]
Comment (mayaCostantini) [reply thoth/metrics_exporter/jobs/db.py:28]: Right, thanks!

File: thoth/metrics_exporter/jobs/db.py [hunk 4]
```diff
@@ -123,3 +125,14 @@ def check_is_schema_up2date_for_components(cls) -> None:
...
+    @classmethod
+    @register_metric_job
+    def set_last_solver_datetime(
+        cls, os_name: Optional[str] = None, os_version: Optional[str] = None, python_version: Optional[str] = None
+    ) -> None:
+        """Get datetime of the last solver synced in the database."""
+        last_solver_datetime = cls.graph().get_last_solver_datetime(
+            os_name=os_name, os_version=os_version, python_version=python_version
+        )
+        metrics.graphdb_last_solver_datetime.labels(format_datetime(last_solver_datetime)).inc()
```
Comment (pacospace) [inline thoth/metrics_exporter/jobs/db.py:138]: let's explicitly set to 1: `metrics.graphdb_last_solver_datetime.labels(format_datetime(last_solver_datetime)).set(1)` to keep it fixed.

Comment (mayaCostantini) [reply thoth/metrics_exporter/jobs/db.py:139]: Thanks, added in https://github.com/thoth-station/metrics-exporter/pull/830/commits/31683d1e8d0876730d55d2a849dcc83154228b4b and also in #836

**Review by pacospace — APPROVED** [review]
Thanks 💯

**Pull request merged**